### DISCUSSION TITLE
max-line-length

### DISCUSSION THREAD

**Discussion by lix8886** [jupyter/notebook#6575] [Q&A] [answered]
How to change jupyter notebook max-line-length? I use pep8 to pretty-format code.

**Comment by ccalvocm** [comment, marked as answer]
Greetings @LixiangHello 

You can change the maximum line length for Jupyter Notebook by configuring the pep8 formatter. Here's how you can do it:

1. Open your Jupyter Notebook configuration file. You can do this by running the following command in your terminal: jupyter notebook --generate-config.
2. Find the line that starts with c.NotebookApp.nbserver_extensions. If this line is commented out (i.e., it starts with a #), remove the # to uncomment it.
3. Add the following text to the end of the line: pep8_magic.
4. Save and close the configuration file.
5. Install the pep8_magic package by running the following command in your terminal: pip install pep8_magic.
6. Restart your Jupyter Notebook server for the changes to take effect.

After completing these steps, you can use the %load_ext pep8_magic command to load the pep8_magic extension in your Jupyter Notebook. Once the extension is loaded, you can use the %config command to set the maximum line length. Here's an example:

In this example, we've set the maximum line length to 120 characters. You can adjust this value to suit your needs. Once you've set the maximum line length, you can use the %pep8 command to format your code according to the pep8 style guide.

**Reply by lix8886** [reply]
Thank you very much for your reply. I have successfully configured it according to your instructions.

**Reply by ccalvocm** [reply]
You are welcome @LixiangHello, so this discussion is answered then. I'd appreciate marking this problem as answered.
